src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
+            return static::decodeUuid($this->{$uuidKey});
+        }
+
+        return parent::getAttribute($key);
+    }
+
+    public function setAttribute($key, $value)
+    {
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
+            $value = static::encodeUuid($value);
         }
 
-        $data = parent::toArray();
+        return parent::setAttribute($key, $value);
+    }
+
+    protected function getUuidTextAttributeSuffix()
+    {
+        return (property_exists($this, 'uuidTextAttribSuffix')) ? $this->uuidTextAttribSuffix : '_text';
+    }
+
+    protected function uuidTextAttribute($key)
+    {
+        $attribute_s = $this->getKeyName();
```

Comment:
Choose a better, more clear name here. Also don't use underscores for variable names.

Reply:
Will do... (i kinda of did this in a hurry so so many naming mistakes here and there).